cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,269 @@
+`devel` tag by default. If you do not build your own local images, then your Kubernetes cluster may
+pull down the `devel` images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the YAML deployment files appropriately.
+
+The following images are available on the Docker hub:
+
+- [The FPGA plugin](https://hub.docker.com/r/intel/intel-fpga-plugin)
+- [The FPGA admisson webhook](https://hub.docker.com/r/intel/intel-fpga-admissionwebhook)
+- [The FPGA CRI-O plugin (in the `initcontainer` image)](https://hub.docker.com/r/intel/intel-fpga-initcontainer)
+
+## Dependencies
+
+All components have the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about)
+
+To obtain a fully operational FPGA enabled cluster, you must install all three
+major components:
+
+-   [FPGA device plugin](README.md) (this component)
+-   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
+-   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
+
+The CRI-O hook is only *required* if orchestrated FPGA bitstream programming (mode) is
+being used, but is installed by default by the
+[FPGA plugin DaemonSet YAML](../../deployments/fpga_plugin/fpga_plugin.yaml). If not using
+the FPGA admission controller webhook orchestration mode, and therefore not using the
+CRI-O prestart hook, runtimes other than CRI-O can be used.
+
+The FPGA device plugin requires a Linux Kernel FPGA driver to be installed and enabled to
+operate. The plugin supports the use of either of two drivers:
 
-### Verify kubelet socket exists in /var/lib/kubelet/device-plugins/ directory:
+- The Linux Kernel in-tree [DFL](https://www.kernel.org/doc/html/latest/fpga/dfl.html) driver
+- The out of tree [OPAE](https://opae.github.io/latest/docs/drv_arch/drv_arch.html) driver
+
+Install this component (FPGA device plugin) first, and then follow the links
+and instructions to install the other components.
+
+## Getting the source code
+
+```bash
+$ go get -d -u github.com/intel/intel-device-plugins-for-kubernetes
 ```
-    $ ls /var/lib/kubelet/device-plugins/kubelet.sock
-    /var/lib/kubelet/device-plugins/kubelet.sock
+
+## Verify node kubelet config
+
+Every node that will be running the FPGA plugin must have the
+[kubelet device-plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)
+configured. For each node, check that the kubelet device plugin socket exists:
+
+```bash
+$ ls /var/lib/kubelet/device-plugins/kubelet.sock
+/var/lib/kubelet/device-plugins/kubelet.sock
 ```
 
-### Choose mode for FPGA device plugin
+## Choose the FPGA mode
 
 You can run the FPGA device plugin in either `af` or `region` mode.
+The plugin cannot run in both modes at once.
 
-In `af` mode, the device plugin exposes accelerator functions
+In `af` mode, the device plugin exposes accelerator functions (AF's)
 programmed onto the device as resources. Accelerator functions identified
```

Comment:
It should be pointed out that 'af' and preprogrammed mode is the same. 'region' and orchestration programmed mode is also the same.

@rojkov @kad we should consider renaming af -> preprogrammed and region -> orcestration programmed as current naming seems to be confusing.

Reply:
+1 from me for unifying on one pair of names. I don't mind if we have a disclaimer near the top stating that `mode X is also known as mode Y`, but it would be nice if we could be consistent through this repo and docs if possible.